tests/integration/controller/kyma/helper_test.go

```diff
@@ -24,12 +25,24 @@ const (
 )
 
 func RegisterDefaultLifecycleForKyma(kyma *v1beta2.Kyma) {
+	mandatoryTemplate := builder.NewModuleTemplateBuilder().
+		WithModuleName("mandatory-template-operator").
+		WithChannel("mandatory").
+		WithMandatory(true).
+		WithOCM(compdescv2.SchemaVersion).Build()
+
 	BeforeAll(func() {
+		Eventually(CreateCR, Timeout, Interval).
```

Comment:
It's better you define a function `DeployMandatoryModuleTemplates`

Reply:
I dont get what this brings us as value 🤔 
The CreateCR function looks like
```
func CreateCR(ctx context.Context, clnt client.Client, obj client.Object) error {
	err := clnt.Create(ctx, obj)
	if !apierrors.IsAlreadyExists(err) {
		return err
	}
	return nil
}
```

And what would the `DeployMandatoryModuleTemplate` do differently? It should also take a ModuleTemplateCR and create it. I do not want to have a function with hardcoded mandatory module templates to stay flexible and want to make use of the templateBuilder

Reply:
what I mean is have a function to make this `BeforeAll` block looks clearly, same pattern as that `DeployModuleTemplates`, `DeleteModuleTemplates`

Then this block can be like this, it's not prevent you reuse `CreateCR`
```
BeforeAll(func() {
  DeployMandatoryModuleTemplates(ctx, controlPlaneClient)
  DeployModuleTemplates(ctx, controlPlaneClient, kyma)
})

func DeployMandatoryModuleTemplates(ctx context.Context, kcpClient client.Client) error {
	mandatoryTemplate := builder.NewModuleTemplateBuilder().
		WithModuleName("mandatory-template-operator").
		WithChannel(mandatoryChannel).
		WithMandatory(true).
		WithOCM(compdescv2.SchemaVersion).Build()
	return CreateCR(ctx, kcpClient, mandatoryTemplate)
}
```

Reply:
ok, will do that